source/dlangbot/github_api.d

```diff
@@ -34,9 +37,50 @@ struct Result
 
 Result ghGetRequest(string url)
 {
-    return request(url, (scope req) {
+    import std.digest.sha : sha1Of;
+    import std.digest : toHexString;
+    import std.file : exists, readText;
+
+    auto cacheFileName = "cache/" ~ cast(string)sha1Of(url).toHexString;
+    Result cacheEntry;
+    if (cacheFileName.exists)
+        cacheEntry = cacheFileName.readText.parseJsonString.deserializeJson!Result();
+
+    auto res = request(url, (scope req) {
         req.headers["Authorization"] = githubAuth;
-    }).expectOK.Result;
+        if (auto p = "ETag" in cacheEntry.headers)
+            req.headers["If-None-Match"] = *p;
+        if (auto p = "Last-Modified" in cacheEntry.headers)
+            req.headers["If-Modified-Since"] = *p;
+    });
+    scope(exit) res.dropBody;
+
+    if (res.statusCode == 304 /*Not Modified*/)
+    {
+        logInfo(" > Cache hit!");
+        return cacheEntry;
+    }
+    else
+    if (res.statusCode == 200 /*OK*/)
```

Comment:
What's with the formatting ?

Reply:
Sorry, what specifically? The `else\nif`?

Reply:
I see this is covered by https://dlang.org/dstyle.html#phobos_whitespace, so I'll fix it.